src/protocol/OpenConnectionReply1.php

```diff
@@ -35,13 +36,26 @@ protected function encodePayload(PacketSerializer $out) : void{
 		$this->writeMagic($out);
 		$out->putLong($this->serverID);
 		$out->putByte($this->serverSecurity ? 1 : 0);
+		if ($this->serverSecurity) {
+			$out->putInt($this->createCookie());
+		}
 		$out->putShort($this->mtuSize);
 	}
 
 	protected function decodePayload(PacketSerializer $in) : void{
 		$this->readMagic($in);
 		$this->serverID = $in->getLong();
 		$this->serverSecurity = $in->getByte() !== 0;
+		if ($this->serverSecurity) {
+			$this->cookie = $in->getInt();
+		}
 		$this->mtuSize = $in->getShort();
 	}
+
+	private function createCookie() : int {
+		$cookieSalt = mt_rand(0, 0xFFFFFFFF);
+		$cookie = Binary::writeLInt($cookieSalt) . Binary::writeLShort(Server::getInstance()->getPort()) . Server::getInstance()->getIp();
```

Comment:
This is not plugin code. You can't import pocketmine\Server here.

Reply:
I think we need to use the client's IP and port but I don't know how to call it.

Reply:
Change the openRequestReply create method to include the cookie?

Reply:
I did. Check it out.